### PR TITLE
Add Bootstrap Confidence Interval Calculation

### PR DESCRIPTION
**Pull request by gwaygenomics**
Modifications to two files:

1. `scripts/util/pancancer_util.R` to include custom bootstrap functions
2. `scripts/viz/ras_summary_figures.R` to include the calls to the bootstrap functions in `dplyr::summarize()`

### CODE REVIEW

File: scripts/util/pancancer_util.R [hunk 1]
```diff
@@ -153,3 +153,40 @@ add_arrow_label <- function(p, x, y, label, offset = c(0, 0, 0, 0)) {
...
+my.mean <- function(x, indices) {
+  # Function to use as `statistic` argument in call to boot()
+  # This will enable a bootstrap estimate of the mean and confidence intervals
+  #
+  # The Function is taken from "Chapter 3 R Bootstrap Examples" by Bret Larget
+  # http://www.stat.wisc.edu/~larget/stat302/chap3.pdf
+  return( mean( x[indices] ) )
+}
+
+get_boot <- function(weight, low = TRUE, R = 1000, min_samples = 5) {
```
Comment (dhimmel) [inline scripts/util/pancancer_util.R]: In general I use [`Hmisc::smean.cl.boot`](https://www.rdocumentation.org/packages/Hmisc/versions/4.0-3/topics/smean.sd), which is what ggplot2 uses. Would this function make it easier?

File: scripts/util/pancancer_util.R [hunk 2]
```diff
@@ -153,3 +153,40 @@ add_arrow_label <- function(p, x, y, label, offset = c(0, 0, 0, 0)) {
...
+  # high_ returned objects.
+  #
+  # Arguments:
+  #    weight: a vector of Ras pathway classifier predictions
+  #    low: boolean of whether to return 95% low or high confidence intervals
+  #    R: integer of how many bootstrap samples to take, defaults to 1000
+  #    min_samples: bootstrapping does not work well with low sample sizes, how
+  #                 many samples are required to calculate a bootstrap estimate
+
+  boot_obj <- boot(weight, statistic = my.mean, R = R)
```
Comment (dhimmel) [inline scripts/util/pancancer_util.R]: This is crazy not to write `boot::boot`.

File: scripts/util/pancancer_util.R (hunk 1 above)
Comment (gwaygenomics) [reply scripts/util/pancancer_util.R]: Looking into this now - thanks for pointing me to the function. I do believe `Hmisc::smean.cl.boot()` is slightly cleaner than a combination of `boot()` and `boot.ci()`. It is also slightly faster and returns a similar result.

```R
> test <- mut_weight_df[mut_weight_df$Hugo_Symbol == 'KRAS', ]
> 
> Hmisc::smean.cl.boot(test$weight)
     Mean     Lower     Upper 
0.8142939 0.8053043 0.8242565 
> boot::boot.ci(
+   boot::boot(test$weight, statistic = my.mean, R = 1000),
+   type = "perc")
BOOTSTRAP CONFIDENCE INTERVAL CALCULATIONS
Based on 1000 bootstrap replicates

CALL : 
boot::boot.ci(boot.out = boot::boot(test$weight, statistic = my.mean, 
    R = 1000), type = "perc")

Intervals : 
Level     Percentile     
95%   ( 0.8043,  0.8247 )  
Calculations and Intervals on Original Scale
> 
> system.time(Hmisc::smean.cl.boot(test$weight))
   user  system elapsed 
  0.012   0.000   0.011 
> system.time(boot::boot.ci(
+   boot::boot(test$weight, statistic = my.mean, R = 1000),
+   type = "perc"))
   user  system elapsed 
  0.016   0.000   0.018
```

The next commit will switch to `Hmisc`

**Comment by gwaygenomics** [comment]
Thanks @dhimmel - I moved to `Hmisc` and the call to `dplyr` is faster. Going to merge!

**Pull request merged**